sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Percentile.scala

```diff
@@ -67,16 +67,23 @@ case class Percentile(
     child: Expression,
     percentageExpression: Expression,
     frequencyExpression : Expression,
+    isIntFreqExpression: Expression,
```

Comment:
This is hacky.
Is this all just so you can pass "0.75" instead of "75"? I don't think that's worth it.

Reply:
@srowen 
If the data has only discrete frequencies, your suggestion is enough (0.75 -> 75)
But, the suggestion isn't enough for fractional number like water with 1/3 L.
1/3 frequency cannot be converted to integer.

Reply:
Oh I misunderstood this a little. This is extending the implementation of weighted percentile.
Still, I doubt you need to add a new expression here. Just expand the allowed types to any `NumericType`, which should be backwards compatible. The math wouldn't change, right?

Reply:
At first, I already implemented it only with NumericType, as you said.

But my first implementation wasn't compatible with hive(hive unittest had failed).
I had found the existing IntegralType implementation which is compatible with hive is only for integer, not rational number.

I couldn't find the way to resolve the problem without additional parameter...

Reply:
Does Hive support this? I didn't see it supports weighted percentile. 
You're also changing the implementation here, right... is it possible that is causing the test to fail without two branches of the logic? I don't think it makes sense to have the caller tell the type of the column; you already know it in the implementation. But I'm wondering how the same calculation can come out differently if it just allows decimal types too. Is it a rounding thing?

Reply:
You are right. The logics are different. So the additional parameter is for logic switching.
I thought the interpolation of value was based on the assumption that weight is integer, which cannot be implemented with floating number. But I can't remember the reason exactly. I'll put concrete example.

Reply:
At the least, I think you can switch logic based on the type of the column, rather than add a new parameter.

I don't know the logic here, but I'd imagine that anything that works for integer weights (1, 2) should work identically for continuous ones (1.0, 2.0). It's possible that the current tests are actually expecting the 'wrong' value if the code is using an unsuitable approximation for integer values or something.

I'd be interested in knowing what fails, as I don't expect Hive-related tests to fail - it doesn't implement weighted percentiles, I thought?